spec/models/post_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  describe 'associations' do
+    it 'should have correct associations' do
+      expect(Post.reflect_on_association(:author).macro).to eq(:belongs_to)
+      expect(Post.reflect_on_association(:comments).macro).to eq(:has_many)
+      expect(Post.reflect_on_association(:likes).macro).to eq(:has_many)
+    end
+  end
+
+  describe 'validations' do
+    it 'should validate presence of name' do
+      post = Post.new(title: nil)
+      expect(post).to_not be_valid
+      expect(post.errors[:title]).to include("can't be blank")
+    end
+
+    it 'should validate numericality of comments_counter' do
+      user = User.create(name: 'name')
+      post = Post.new(title: 'title', text: 'text', author_id: user.id, comments_counter: -1)
+      expect(post).to_not be_valid
+      expect(post.errors[:comments_counter]).to include('must be greater than or equal to 0')
+    end
+    it 'should validate numericality of likes_counter' do
+      user = User.create(name: 'name')
+      post = Post.new(title: 'title', text: 'text', author_id: user.id, likes_counter: -1)
+      expect(post).to_not be_valid
+      expect(post.errors[:likes_counter]).to include('must be greater than or equal to 0')
+    end
+  end
```

Comment:
- [ ] You only wrote tests that don't show a model instance can't be valid. Based on the requirements, you also need to be sure that the validations you set work properly when a model instance has all the values right. It also ensures that data inputted into the database is consistent and helps to prevent unexpected issues in your application. Kindly write tests that show the presence of the validation or show that the model instances are valid when created properly. 😎 💻